src/portable/instance/backup.rs

```diff
@@ -203,6 +238,11 @@ pub async fn restore(cmd: &Restore, opts: &crate::options::Options) -> anyhow::R
         return Ok(());
     }
 
+    if let InstanceName::Local(inst_name) = &cmd.instance {
+        let inst_name = inst_name.clone();
+        tokio::task::spawn_blocking(move || super::control::do_stop(&inst_name)).await??;
```

Comment:
We want to notify the user that the instance will be stopped (or maybe even confirm that?)

Reply:
We currently back up and restore the whole DB, but I'll make sure this is worded properly.

Reply:
```rust

    let stop_warning = if let InstanceName::Local(_) = &cmd.instance {
        "This will stop the instance and restore all branches from the backup. Any data not backed up will be lost."
    } else {
        "This will restore all branches from the backup. Any data not backed up will be lost."
    };

    let prompt = format!(
        "Will restore {inst_name:#} from the specified backup. {stop_warning}\
        \n\nContinue?",
    );

```

Reply:
> "This will stop the instance and restore all branches from the backup. Any data not backed up will be lost."

I'd also add roughly this: "... After the restore operation is completed, the instance will be restarted."

BTW, can restore theoretically error out? Should we spell that out or the probability is so low that we don't want to?

Reply:
The restore _could_ fail but we perform the entire restore in a tempdir and then do two renames to get it into the right spot. Basically we only do the final swap of old and new dir once pg_verifybackup confirms it's valid.